test/com/google/javascript/jscomp/parsing/TypeSyntaxTest.java

```diff
@@ -111,6 +111,16 @@ public void testTypeInDocAndSyntax() {
     parse("var /** string */ foo: string = 'hello';");
   }
 
+  public void testTypedGetterSetterDeclaration() {
+    Node n = parse("var x = {get a(): number {\n}};", LanguageMode.ECMASCRIPT6_TYPED);
+    assertDeclaredType("number type", numberType(),
+        n.getFirstChild().getFirstChild().getFirstChild().getFirstChild().getFirstChild());
+    n = parse("var x = {set a(v: number) {\n}};", LanguageMode.ECMASCRIPT6_TYPED);
+    assertDeclaredType("number type", numberType(),
+        n.getFirstChild().getFirstChild().getFirstChild().getFirstChild().getFirstChild().getChildAtIndex(1)
```

Comment:
Can we add one more test that we get a parse error for setters with a return?
e.g.
  var x = {set a(x): number {}}

Also, if it does work, consider adding a test to NewTypeInferenceES6TypedTest.java, such as the following:
    typeCheck(
        "var x = {get a(): number { return 'str'; }}",
        NewTypeInference.RETURN_NONDECLARED_TYPE);

(I consider this a more straightforward way of checking that the types are correctly parsed than traversing the AST)